### PR TITLE
Add SWM4-NDP water to modeller.addSolvent()

### PR DESCRIPTION
**Pull request by andrewsb8**
Hi all,

I opened issue #3415 earlier this year regarding the inability to load a pdb of a protein and then use OpenMM to solvate the system with this water for use with polarizable force fields like the CHARMM Drude family. The current solution is to solvate with another force field and tip3p water, load the polarizable force field, and add the extra particles.

This could be confusing to newer users (me) and also has some weird consequences. For example, as discussed at the end of #3415, I was attempting to simulate a tripeptide glycine isoleucine glycine with charged N and neutral C termini. Since I had to solvate the tripeptide with a different force field (CHARMM36m), the neutral C terminus atom naming in the C36m is different than for Drude, leading to the following after trying to load parameters for the Drude force field for the solvated system.

`ValueError: No template found for residue 3 (GLY). The set of atoms is similar to SER, but it is missing 12 atoms.`

This simple fix should simplify this process and help avoid unnecessary steps, like manual modifications of pdbs between preparation steps, for users to easily utilize the force fields available in the suite. Here is what I did:

1. Loaded the cubic box with 3 nm edge of tip3p water included in openmm/wrappers/python/openmm/app/data and converted it to swm4-ndp with the following:

````
pdb = PDBFile("./tip3p.pdb") #tip3p box pdb file included with OpenMM

modeller = Modeller(pdb.topology, pdb.positions)
forcefield = ForceField("charmm_polar_2019.xml")
modeller.addExtraParticles(forcefield) #convert tip3p water to swm4-ndp water in box
````

2. Ran a production simulation (after energy minimzation) for 20 ns at 300K with a time step of 0.5 fs and tracked various properties to test for equilibration. The temperature does vary between 298 and ~315K which might indicate my time step was still slightly too large. But everything looks ok otherwise:

swm4-ndp-log

3. Add a pdb of this box to OpenMM: openmm/wrappers/python/openmm/app/data/swm4ndp.pdb which will be called by `modeller.addSolvent()`

4. Added the option for this water model to be called by `addSolvent` and edited the comments at the beginning of the function

````
elif model == 'swm4ndp':
            waterRadius = 0.318395*vdwRadiusPerSigma
````
Where the sigma value of 0.318395 was taken from Table 1 of Lamoureux Jour. Phys. Chem. Let. 2006, the paper introducing this water model.

5. Installed my fork from source on a fresh install of Ubuntu 22.04 and solvated the same GIG tripeptide in a cubic box with 5 nm edges using the following:

```
modeller = Modeller(pdb.topology, pdb.positions)
forcefield = ForceField('charmm_polar_2019.xml')
modeller.addExtraParticles(forcefield)
modeller.addSolvent(forcefield, boxSize=Vec3(5, 5, 5)*nanometers, neutralize=True, negativeIon='Cl-', positiveIon='Na+', model='swm4ndp')
```

Here is the pdb of the sovlated (not minimized) structure: [solvated-fixed.pdb.txt](https://github.com/openmm/openmm/files/9465055/solvated-fixed.pdb.txt). It started minimizing the energy but I was testing on a virtual machine where GPU passthrough is not available. So, I did not run a long production simulation with these changes as the CPU platform on a 1 CPU VM would take a looong time. 

I did not write any automated tests for this change and the New Feature Checklist appears to be blank (https://github.com/openmm/openmm/wiki/Checklist-when-adding-a-new-feature). I would be happy to write some tests if needed. I understand this is a micro change for a pull request but I'm happy to write some tests for this or make other changes if needed. Thanks for reading.

Brian

### CODE REVIEW

**Comment by peastman** [comment]
Thanks!  This looks good.  Can you update `test_addSolventWaterModels()` in `TestModeller.py` to make it check this water model?  That should be good for automated tests, and we can manually do a few sanity checks to make sure it simulates correctly.

Where did you get that URL for the new feature checklist from?  The correct URL is https://github.com/openmm/openmm/wiki/Checklist-for-Adding-a-New-Feature.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 __pycache__
 build
 build?
+fork-notes
```
Comment (peastman) [inline .gitignore:4]: Can you remove this line?  I don't think it was meant to be included.

Comment (andrewsb8) [reply .gitignore]: Done

**Comment by andrewsb8** [comment]
I added the water model to `test_addSolventWaterModels()`. 

I got the link from CONTRIBUTING.md (https://github.com/openmm/openmm/blob/master/CONTRIBUTING.md) by clicking on this hyperlink.

Screenshot from 2022-09-02 10-04-18

**Comment by peastman** [comment]
I just fixed the URL in #3764.  Thanks for pointing that out!

The updated test case is passing.  Is this still a work in progress, or is it ready to merge?

**Comment by andrewsb8** [comment]
No problem! It's ready to merge.

**Comment by peastman** [comment]
Thank you for your contribution!

**Comment by andrewsb8** [comment]
Thanks for accepting it! Packaging software like this to be available through managers like conda is outside of my wheel house. I know if I compile master from source this change will be installed.

Can I perform a conda update and see these changes now that it's merged? Or is there a lag between the merge and package update?

**Comment by peastman** [comment]
It will be in the next release.  We're trying to get that out, but running into lots of problems building packages.  I *hope* we'll have a beta available very soon.

**Pull request merged**